src/features/ai/settings/AiSettingsPane.tsx

```diff
@@ -52,10 +66,42 @@ export function OpenAiSettingsPane({ config }: SettingsPaneProps) {
     saveMutation.mutate(paneSettings);
   };
 
-  const isDirty = JSON.stringify(paneSettings) !== JSON.stringify(getOpenAISettingsCached());
+  const isDirty = JSON.stringify(paneSettings) !== JSON.stringify(getAiSettingsCached());
 
   return (
     <SettingsPane config={config} header={config.title} isDirty={isDirty} onSave={handleSaveSettings}>
+      <div className="flex flex-col items-start gap-2">
+        <h2 className="t text-modal-txt-primary">Provider</h2>
+        <Dropdown
+          aria-label="provider"
+          data-testid={MODEL_PROVIDER_TEST_ID}
+          options={MODEL_PROVIDER_OPTIONS.map((option) => ({
+            name: option.label,
+            value: option.provider,
+          }))}
+          value={paneSettings.provider}
+          onChange={(provider: ModelProvider) => setPaneSettings({ ...paneSettings, provider })}
+        />
+      </div>
+
+      <div className="flex flex-col items-start gap-2">
+        <h2 className="text-modal-txt-primary">Provider Model</h2>
+        <Input
+          data-testid={MODEL_TEST_ID}
+          value={paneSettings.model}
+          onChange={(chat_model) => setPaneSettings({ ...paneSettings, model: chat_model })}
+        />
+        <InfoText>
+          <strong>Use: </strong>
+          <InfoTextOptions
+            options={[
+              { value: 'gpt-3.5-turbo', label: 'OpenAI' },
+              { value: 'llama2', label: 'for Ollama' },
+            ]}
+            onClick={(value) => setPaneSettings({ ...paneSettings, model: value })}
+          />
```

Comment:
My understanding of the backend code is that the model is overridden, to use 'llama2' when the provider is ollama. Maybe we should reflect that in the frontend by disabling the field and displaying 'llama' when selecting ollama provider?

Reply:
I will make the llama2 default when we select Ollama and same for OpenAI. They can use other values like gpt4 or other ollama models.